src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -777,22 +830,38 @@ protected virtual bool TryLoadFromCache(JsonOperationContext context)
                 }
                 catch (HttpRequestException e) // server down, network down
                 {
-                    if (shouldRetry == false)
-                        throw;
+                    using (GetContext(out var requestContext))
+                    {
+                        if (e.InnerException is SocketException se && ShouldRemoveHttpClient(se))
```

Comment:
Also, probably for proxy errors, no?

Reply:
I was getting `SocketError.ConnectionRefused` in that case (handled). Do you know if there is anything else?

Reply:
Consider the case where the proxy is alive, but returning errors to us. 
Proxy auth failure, for example.

Reply:
Not sure that I entirely follow. I should retry on 407?

Reply:
Yes

Reply:
I will handle 407 in a separate PR